motif/core.py

```diff
@@ -169,43 +182,76 @@ def get_labels(self, annotation_fpath, overlap_threshold=0.5):
 
         '''
         annot_times, annot_freqs = _load_annotation(annotation_fpath)
+        ref_times, ref_cent, ref_voicing = _format_annotation(
+            annot_times, annot_freqs, self.duration, self.sample_rate
+        )
+        est_cents, est_voicing = _format_contour_data(self.freqs)
+
         labels = dict.fromkeys(self.nums)
         overlaps = dict.fromkeys(self.nums)
 
         for i in self.nums:
-            this_contour_times = self.contour_times(i)
-            this_contour_freqs = self.contour_freqs(i)
-            start_t = this_contour_times[0]
-            end_t = this_contour_times[-1]
-            gt_idx = np.logical_and(
-                annot_times >= start_t, annot_times <= end_t
-            )
-            this_annot_times = annot_times[gt_idx]
-            this_annot_freqs = annot_freqs[gt_idx]
-            res = mir_eval.melody.evaluate(
-                this_annot_times, this_annot_freqs,
-                this_contour_times, this_contour_freqs
+            gt_idx = _get_snippet_idx(self.contour_times(i), ref_times)
+            overlaps[i] = mir_eval.melody.overall_accuracy(
+                ref_voicing[gt_idx], ref_cent[gt_idx],
+                est_voicing[self.index_mapping[i]],
+                est_cents[self.index_mapping[i]]
             )
-            overlaps[i] = res['Overall Accuracy']
-            labels[i] = 1*(overlaps[i] > overlap_threshold)
+            labels[i] = 1 * (overlaps[i] > overlap_threshold)
```

Comment:
why do you multiply by 1? I'm not aware of this magic to convert from bools to ints.